bin/fuzz.py

```diff
@@ -188,16 +187,25 @@ def run(self):
         self.tasks = 0
         self.queue = parallel.Parallel(lambda task: task.run(bar), pin=True)
 
-        # pool of ids used for generators
-        self.tmp_ids = 2 * max(1, self.queue.num_threads) + 1
-        self.free_tmp_id = {*range(self.tmp_ids)}
-        self.tmp_id_count = [0] * self.tmp_ids
+        if self.queue.num_threads:
+            # pool of ids used for generators
+            self.tmp_ids = 2 * max(1, self.queue.num_threads) + 1
+            self.free_tmp_id = {*range(self.tmp_ids)}
+            self.tmp_id_count = [0] * self.tmp_ids
+
+            # add first generator task
+            self.finish_task()
 
-        # add first generator task
-        self.finish_task()
+            # wait for the queue to run empty (after config.args.time)
+            self.queue.join()
+        else:
+            self.tmp_ids = -1
```

Comment:
Comment here that this is the single-threaded case. Maybe this should also include num_threads=1? Not sure.

Reply:
`num_threads=1` should work fine since we have one thread (besides the main thread locked by `join`) that handles elements in the queue